src/services/ApiService/History/types.ts

```diff
@@ -0,0 +1,27 @@
+import { ITxStatus, TAddress, ITxHash, ITxType } from '@types';
+
+export interface ITxHistoryApiResponse {
+  readonly blockNumber?: number;
+  readonly data: string;
+  readonly erc20Transfers?: ITxHistoryERC20Transfer[];
+  readonly from: TAddress;
+  readonly gasLimit: number;
+  readonly gasPrice: number;
+  readonly gasUsed?: number;
+  readonly hash: ITxHash;
+  readonly nonce: number;
+  readonly recipientAddress: TAddress;
+  readonly status: ITxStatus.PENDING | ITxStatus.SUCCESS | ITxStatus.FAILED | ITxStatus.UNKNOWN;
+  readonly timestamp?: number;
+  readonly to: TAddress;
+  readonly value: number;
+
+  readonly txType: ITxType;
+}
+
+export interface ITxHistoryERC20Transfer {
+  readonly from: TAddress;
+  readonly to: TAddress;
+  readonly contractAddress: TAddress;
+  readonly amount: number;
+}
```

Comment:
Some of these props are of type `number`. I'm surprised this doesn't toss any overflow/underflow issues with the BN-esque values the api returns. Is this a potential problem?

Reply:
We need to solve this in the API, reached out on Slack.